src/context/Product-context.js

```diff
@@ -10,123 +10,119 @@ const useProducts = () => useContext(ProductContext);
 const ProductContextProvider = ({children}) => {
     const [loading , setLoading] = useState(true);
     const [data , setData] = useState([]);
-    
 
     const highToLowPrice = (a,b) => {
        return [b.price - a.price];
     }
 
-const lowToHighPrice = (a,b) => {
-  return [a.price - b.price];
-}
+    const lowToHighPrice = (a,b) => {
+       return [a.price - b.price];
+    }
 
 //   Sorting Filter
-const sortedData = () => {
-    let sortData = [...data]
+const sortedData = (products) => {
     if(state.payload === 'high_to_low'){
-        return [...sortData.sort(highToLowPrice)]
+        return [...products.sort(highToLowPrice)]
     }
     else if(state.payload === 'low_to_high'){
-      return [...sortData.sort(lowToHighPrice)]
+      return [...products.sort(lowToHighPrice)]
     }
     else if(state.payload >= 10000){
-        sortData = sortData.filter(product => {
+        products = products.filter(product => {
            return parseInt(product.price , 10) <= parseInt(state.payload , 10)}) 
+   }else{
+       return products
    }
-return sortData;
+return products;
 }
 
   //  Ratings Filter
-  const ratingFilterData = () => {
-   let ratedData = sortedData();
+  const ratingFilterData = (products) => {
+   
    if(state.type === '4STAR'  && state.payload === true){
-    ratedData = ratedData.filter(product => {
+    return products.filter(product => {
            return parseInt(product.rating) > 4
        })
    }else if(state.type === '3STAR'  && state.payload === true){
-    ratedData = ratedData.filter(product => {
+    return products.filter(product => {
           return parseInt(product.rating) > 3
       })
   }else if(state.type === '2STAR'  && state.payload === true){
-    ratedData = ratedData.filter(product => {
+    return products.filter(product => {
           return parseInt(product.rating) > 2
       })
   }else if(state.type === '1STAR'  && state.payload === true){
-    ratedData = ratedData.filter(product => {
+    return products.filter(product => {
           return parseInt(product.rating) > 1
       })
   }else{
-    ratedData = ratedData.filter(product => {
-          return product
-      })
+     return products
   }
-
-  return ratedData;
   }
 
   // Brands Filter
-  const brandFilterData = () => {
-      let brandData = ratingFilterData();
+  const brandFilterData = (products) => {
```

Comment:
can create a utils folder: to write all repetitive logic ==> chance are high same methord will use in wishlist , cart , product page